autowire/shared/src/main/scala/autowire/Macros.scala

```diff
@@ -79,61 +80,61 @@ object Macros {
           None
         }
       }
-      val argName = c.freshName[TermName](TermName("args"))
-      val args: Seq[Tree] = flattenedArgLists.zipWithIndex.map { case (arg, i) =>
+      val argName: TermName  = c.freshName[TermName](TermName("args"))
+      val args   : Seq[Tree] = flattenedArgLists.zipWithIndex.map { case (arg, i) =>
         val default = hasDefault(i) match {
           case Some(defaultName) => q"scala.util.Right(($target).${TermName(defaultName)})"
-          case None => q"scala.util.Left(autowire.Error.Param.Missing(${arg.name.toString}))"
+          case None              => q"scala.util.Left(autowire.Error.Param.Missing(${arg.name.toString}))"
         }
         q"""autowire.Internal.read[$pickleType, ${arg.typeSignature}](
-                 $argName,
-                 $default,
-                 ${arg.name.toString},
-                 ${c.prefix}.read[${arg.typeSignature}](_)
-               )
-             """
+              $argName,
+              $default,
+              ${arg.name.toString},
+              ${c.prefix}.read[${arg.typeSignature}](_)
+            )
+         """
       }
 
-
-      //val memSel = c.universe.newTermName(memPath.mkString("."))
-
-      val bindings = args.foldLeft[Tree](q"Nil") { (old, next) =>
+      val bindings: Tree = args.foldLeft[Tree](q"Nil") { (old, next) =>
         q"$next :: $old"
       }
 
-      val nameNames: Seq[TermName] = flattenedArgLists.map(x => x.name.toTermName)
-      val assignment = flattenedArgLists.foldLeft[Tree](q"Nil") { (old, next) =>
+      val params    : Seq[TermName] = flattenedArgLists.map(x => x.name.toTermName)
+      val assignment: Tree          = flattenedArgLists.foldLeft[Tree](q"Nil") { (old, next) =>
         pq"scala.::(${next.name.toTermName}: ${next.typeSignature} @unchecked, $old)"
       }
 
-
-
-      val memSel = memPath.foldLeft(q"($target)") { (cur, nex) =>
-        q"$cur.${c.universe.newTermName(nex)}"
+      val memSel: Tree = memPath.foldLeft(q"($target)") { (cur, nex) =>
+        q"$cur.${TermName(nex)}"
       }
 
-      val futurized = futurize(q"$memSel(..$nameNames)", meth)
-      val fullPath = prefixPath ++ memPath
-      val frag = cq""" autowire.Core.Request(Seq(..$fullPath), $argName) =>
-             autowire.Internal.doValidate($bindings) match{ case (..$assignment) =>
-               $futurized.map(${c.prefix}.write(_))
-               case _ => ???
-             }
-           """
+      val futurized: Tree = if (params.isEmpty) {
+        // Accessor method
+        futurize(q"$memSel", meth)
```

Comment:
Method invocation without parameter list allows for accessor methods.

If a method is defined with empty parenthesis `def someMethod()` it will be materialised as an accessor method `def someMethod` which seems to be fine (all tests pass with/without parenthesis).